### PR TITLE
feat: `forest-cli archive info`

### LINKED ISSUE #3166: `forest-cli archive info`
Command-line tool for printing basic information about CAR snapshots.

- [x] CAR variant: v1, v2, indexed-v1
- [x] Network (calibnet, mainnet)
- [x] Epoch
- [x] Available tipsets
- [x] Available messages

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Implement `forest-cli archive info` which displays basic information about an archive file.

## Reference issue to close (if applicable)

Closes #3166 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/cli/subcommands/archive_cmd.rs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+//! Archives are key-value pairs encoded as
+//! [CAR files](https://ipld.io/specs/transport/car/carv1/). The key-value pairs
+//! represent a directed, acyclic graph (DAG). This graph is often a subset of a larger
+//! graph and references to missing keys are common.
+//!
+//! Each graph contains blocks, messages, state trees, and miscellaneous data
+//! such as compiled `WASM` code. The amount of data differs greatly in different
+//! kinds of archives. While there are no fixed definitions, there are three
+//! common kind of archives:
+//!   A full archive contains a complete graph with no missing nodes. These
```
Comment (LesnyRumcajs) [inline src/cli/subcommands/archive_cmd.rs:13]: It might be more readable to use bullet points than just indentation.

**Review by LesnyRumcajs — COMMENTED** [review]
I like it. Shall we add some tests (that is, ones calling the binary so it goes through the whole execution path)? Perhaps a short guide to the `developer documentation`?

**Comment by lemmih** [comment]
> I like it. Shall we add some tests (that is, ones calling the binary so it goes through the whole execution path)? Perhaps a short guide to the `developer documentation`?

It's slightly awkward adding those tests now since the command only accepts uncompressed CAR files. Let's wait until we have directly support for loading compressed archives (Aatif will soon have a proof-of-concept ready). Once that is done, the different commands (`snapshot export`, `archive info`, `archive export`, `snapshot validate`, etc) will all read and write the same formats, which makes testing easier.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/cli/subcommands/archive_cmd.rs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                    } else if genesis_cid.to_string() == mainnet::GENESIS_CID {
+                        network = "mainnet".into();
+                    }
+                    break;
+                }
+            }
+        }
+
+        Ok(ArchiveInfo {
+            variant: "CARv1".into(),
```
Comment (creativcoder) [inline src/cli/subcommands/archive_cmd.rs:178]: I wonder if this variant info can be made available within car archive; wouldn't need to hardcode it then if we want to support versioning in future

Comment (lemmih) [reply src/cli/subcommands/archive_cmd.rs:178]: Determining whether the content of a file is in the CARv1 or CARv2 format is up to us. The formats are not compatible and have to be treated differently.

**Pull request merged**